src/app/cli/src/cli_parser.rs

```diff
@@ -1244,6 +1244,39 @@ pub fn cli() -> Command {
                                     .required(true)
                                     .help("An existing GitHub access token"),
                             ]),
+                        Command::new("compact")
+                            .about("Compact a dataset")
+                            .args([
+                                Arg::new("dataset")
+                                    .action(ArgAction::Append)
+                                    .index(1)
+                                    .required(true)
+                                    .value_parser(value_parse_dataset_ref_local)
+                                    .help("Local dataset reference(s)"),
+                                Arg::new("max-slice-size")
+                                    .long("max-slice-size")
+                                    .value_parser(value_parser!(u64))
+                                    .default_value("0")
+                                    .value_name("SIZE")
+                                    .help("Maximum size of a single data slice file in bytes"),
+                                Arg::new("hard")
+                                    .long("hard")
+                                    .action(ArgAction::SetTrue)
+                                    .help("Run a hard dataset compaction"),
```

Comment:
`Perform 'hard' compaction that rewrites the history of a dataset`

Reply:
Changed